### PR TITLE
Use new controller system for extension controller

### PR DESCRIPTION
**Pull request by imkingdavid**
This depends on the currently-unmerged phpbb3 PR #995. I suggest you don't merge this in until that is merged. However, I am sending the PR so that people can see how to use the new system to create a controller for extensions.

Note that creating a controller for the core is an identical process. Controllers should be contained in a new folder in the root called /controller/ and the routing and services should go in the respective files located in the core /config/ directory.

Controllers are accessed in the URL via ./phpBB/app.php/pattern (where pattern matches the pattern defined in the routing file for the controller you are trying to access).

### CODE REVIEW

**Comment by michaelcullum** [comment]
Update the readme please

**Comment by igorw** [comment]
@naderman you can merge this now

**Comment by imkingdavid** [comment]
Actually I haven't been keeping this up to date lately. Let me make sure it is all correct.

**Comment by igorw** [comment]
@imkingdavid status? it's kind of important to have this up to date.

File: controller/main.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	}
+
+	/**
+	* Foo controller to be accessed with the URL is /ex
+	* (where {test} is the placeholder for a value)
+	*
+	* @param string $test String taken from the URL
+	* @return Symfony\Component\HttpFoundation\Response A Symfony Response object
+	*/
+	public function bar($test = bar)
```
Comment (igorw) [inline controller/main.php]: parse error.

File: controller/main.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	* NOTE: The parameters of this method must match in order and type with
+	* the dependencies defined in the services.yml file for this service.
+	*
+	* @param phpbb_request $request Request object
+	* @param phpbb_user $user User object
+	* @param phpbb_template $template Template object
+	* @param dbal $db DBAL object
+	* @param phpbb_controller_helper $helper Controller helper object
+	*/
+	public function __construct(phpbb_request $request, phpbb_user $user, phpbb_template $template, dbal $db, phpbb_controller_helper $helper)
```
Comment (marc1706) [inline controller/main.php]: `dbal $db` should be `phpbb_db_driver $db` or else you'll get a fatal error.

**Comment by imkingdavid** [comment]
This should be good to go, I think

**Pull request closed without merging**